rust/noosphere-name-system/src/cli.rs

```diff
@@ -0,0 +1,152 @@
+use anyhow::{anyhow, Result};
+use clap::{Parser, Subcommand};
+use futures::future::try_join_all;
+use noosphere::authority::generate_ed25519_key;
+use noosphere::authority::restore_ed25519_key;
+use noosphere_cli::native::workspace::Workspace;
+use noosphere_name_system::NameSystem;
+use std::fs;
+use tokio;
+use tokio::signal;
+use toml;
+/// @TODO these materials should be exposed in noosphere::authority
+use ucan_key_support::ed25519::Ed25519KeyMaterial;
```

Comment:
@cdata WDYT about exposing key details in `noosphere::authority`? Though more UCAN integration will be needed here (in name-system, not specifically this CLI dev shell)

Reply:
There are some generalized helpers that we use across the crates for generating and managing these keys:

https://github.com/subconsciousnetwork/noosphere/blob/88b08e3dc6e9df8386ed109b81bb38c8f06ceb4f/rust/noosphere/src/authority/key_material.rs#L16-L38

I would be happy for us to expand on these. Although, I would shy away from re-exporting implementation directly from other crates unless it seems like we really need to.